terraform/cloudwatch_dashboard.tf

```diff
@@ -0,0 +1,291 @@
+// CloudWatch dashboard for BinaryAlert
+
+locals {
+  /* Ideally, each widget could be described with Terraform objects instead of raw JSON, but
+     Terraform does not yet support complex types (in particular, lists of maps are not allowed). */
+
+  s3_bucket_stats = <<EOF
+{
+  "type": "metric",
```

Comment:
I would prefer to use variables to store values of `type`, `width`, `height`.

Reply:
The `type`, `width`, and `height` could be different for each widget. For example, the first two widgets have `height: 3` and the rest default to `height: 6`. It happens that all of our widgets right now have `width: 12` and `type: "metric"`, but that could change.

By `variables` did you mean exposing variables in `variables.tf` to the user? I'm not sure what could be user-configurable. For example, the width is out of 24 max "units"; setting it to 12 just means it will take up half of the width of the dashboard.